pkg/apis/serving/k8s_validation.go

```diff
@@ -105,8 +106,11 @@ func ValidateVolumes(ctx context.Context, vs []corev1.Volume, mountedVolumes set
 		}
 		errs = errs.Also(validateVolume(ctx, volume).ViaIndex(i))
 		volumes.Insert(volume.Name)
+		if volume.EmptyDir != nil {
+			nonReadOnlyVolumes.Insert(volume.Name)
```

Comment:
the variable is called `nonReadOnlyVolumes`, but it seems like it actually contains nonEmptyDirVolumes. When we actually use it we end up negating it again (`!nonReadOnlyValumes.Has(..)`) which is a lot of negations for my mind to follow and confused me a bit: maybe this could be `emptyDirVolumes` or `allowedReadWriteVolumes` or something like that?

Reply:
(see suggested alternative approach in 3rd comment though which makes this comment and the next redundant if it works!)

Reply:
I can change the name but the var contains only emptyDirs.